dask/dataframe/io/parquet/core.py

```diff
@@ -911,6 +912,18 @@ def get_engine(engine):
         return eng
 
     elif engine in ("pyarrow", "arrow", "pyarrow-legacy", "pyarrow-dataset"):
+
+        pa = import_required("pyarrow", "`pyarrow` not installed")
+        pa_version = parse_version(pa.__version__)
+
+        if engine in ("pyarrow", "arrow"):
+            engine = "pyarrow-dataset"
+        elif pa_version.major >= 5 and engine == "pyarrow-legacy":
+            warnings.warn(
+                "`ArrowLegacyEngine` ('pyarrow-legacy') is deprecated "
+                "for pyarrow>=5. Please use `engine='pyarrow'`."
+            )
```

Comment:
Nitpick: I think this will raise a `UserWarning` by default. It would be more accurate if we emitted a `FutureWarning` instead